src/main/java/seedu/address/model/colabfolderhistory/SavedState.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.colabfolderhistory;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.model.ReadOnlyColabFolder;
+
+/**
+ * Encapsulates information regarding the state of the {@code ColabFolder} at a certain point in time.
+ */
+public class SavedState {
+    private final ReadOnlyColabFolder colabFolder;
+    private final CommandResult commandResult;
+
+    /**
+     * Constructs a {@code SavedState} object with a {@code colabFolder} and {@code commandResult}.
+     *
+     * @param colabFolder   The {@code ReadOnlyColabFolder} to store, should not be null.
+     * @param commandResult The {@code CommandResult} to store.
+     */
+    public SavedState(ReadOnlyColabFolder colabFolder, CommandResult commandResult) {
```

Comment:
This one I feel it would be better to have 2 constructors. 1 that has only  1 parameter for ColabFolder and sets CommandResult to null be default. And another constructor that has 2 parameters such that both parameters would have requireNonNull.

Not a very big deal because the Null in parameter is only used in ColabFolderHistory

CommandResult.